datafusion/functions/src/string/split_part.rs

```diff
@@ -105,13 +108,70 @@ impl ScalarUDFImpl for SplitPartFunc {
     }
 }
 
-macro_rules! process_split_part {
-    ($string_array: expr, $delimiter_array: expr, $n_array: expr) => {{
-        let result = $string_array
-            .iter()
-            .zip($delimiter_array.iter())
-            .zip($n_array.iter())
-            .map(|((string, delimiter), n)| match (string, delimiter, n) {
+/// Splits string at occurrences of delimiter and returns the n'th field (counting from one).
+/// split_part('abc~@~def~@~ghi', '~@~', 2) = 'def'
+pub fn split_part<StringArrayLen: OffsetSizeTrait, DelimiterArrayLen: OffsetSizeTrait>(
+    args: &[ArrayRef],
+) -> Result<ArrayRef> {
+    let n_array = as_int64_array(&args[2])?;
+
+    match (args[0].data_type(), args[1].data_type()) {
+        (DataType::Utf8View, DataType::Utf8View) => {
+            split_part_impl::<&StringViewArray, &StringViewArray, StringArrayLen>(
+                args[0].as_string_view(),
+                args[1].as_string_view(),
+                n_array,
+            )
+        }
+        (_, DataType::Utf8View) => split_part_impl::<
```

Comment:
I almost wonder if it would be simpler to call `spit_part_impl` directly from `invoke`

This extra level of indirection does save some repetition, but it might be clearer if all the possible combination of argument types were simply specified directly